### PR TITLE
forward_command_controller

### LINKED ISSUE #78: Implement forward_command_controller from ROS1
See https://github.com/ros-controls/ros_controllers/tree/melodic-devel/forward_command_controller/include/forward_command_controller.

This is required by #50, #51 and #52.

Also see https://github.com/ros-controls/ros2_control/issues/124.

### PR DESCRIPTION
**Pull request by jordan-palacios**
Port of forward_command_controller from ROS1.

Depends https://github.com/ros-controls/ros2_controllers/pull/90.

Closes #78.

### CODE REVIEW

**Comment by AndyZe** [comment]
@jordan-palacios can I help you with this? Testing/ debugging/ coding ... whatever you need.

**Comment by jordan-palacios** [comment]
> @jordan-palacios can I help you with this? Testing/ debugging/ coding ... whatever you need.

Thanks @AndyZe but this PR it's pretty much done. Right now is blocked until https://github.com/ros-controls/ros2_controllers/pull/90 is merged.

**Comment by AndyZe** [comment]
Great! I will make it my goal to get this running in simulation with fake_joint_driver, then

File: forward_command_controller/src/forward_command_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+          rclcpp::get_logger(
+            logger_name_), "joint '" << joint_name << "' not registered");
+        return CallbackReturn::ERROR;
+      }
+    }
+
+    // get joint handles
+    for (const auto & joint_name : joint_names) {
+      hardware_interface::JointHandle joint_handle(joint_name, interface_name);
+      if (rh_ptr->get_joint_handle(joint_handle) ==
```
[the remark this answers is not in the document]
Comment (jordan-palacios) [reply forward_command_controller/src/forward_command_controller.cpp:84]: Problem with `get_joint_handles` is that returns _all_ available handles and we may want a subset here. Maybe we could add a `get_joint_handles` that also takes a list of names `joint_names` argument to `robot_hardware`?

File: forward_command_controller/src/forward_command_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+#include <string>
+#include <utility>
+
+#include "forward_command_controller/forward_command_controller.hpp"
+#include "rclcpp/qos.hpp"
+#include "rclcpp/logging.hpp"
+
+namespace
+{
+constexpr auto kFCCLoggerName = "forward command controller";
```
[the remark this answers is not in the document]
Comment (jordan-palacios) [reply forward_command_controller/src/forward_command_controller.cpp:25]: The logger name is already a member variable, though non const so the controllers that will inherit from this one may change it. I could make it to be default initialized to `"forward command controller"` and remove the `kFCCLoggerName` altogether. 

What do you think?

[the remark this answers is not in the document]
Comment (bmagyar) [reply forward_command_controller/src/forward_command_controller.cpp:25]: me gusta option 2

File: forward_command_controller/src/forward_command_controller.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bmagyar) [reply forward_command_controller/src/forward_command_controller.cpp:84]: you are right! let's not dwell on this one, I put it on the top shelf in my brain

File: forward_command_controller/src/forward_command_controller.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jordan-palacios) [reply forward_command_controller/src/forward_command_controller.cpp]: Done.

**Review by bmagyar — APPROVED** [review]

**Pull request merged**